app/presenters/candidate_interface/application_form_presenter.rb

```diff
@@ -44,8 +44,46 @@ def section_errors
         .map(&:first)
     end
 
+    ApplicationChoiceError = Struct.new(:message, :course_choice_id) do
+      def anchor
+        "#course-choice-#{course_choice_id}"
+      end
+    end
+
+    def application_choice_errors
```

Comment:
We've been trying to shake off the need for presenters, could this logic also live inside the component itself?

Reply:
Possibly, although seems like it'd be a significant refactor, as it'd entail introducing a new parent component or modifying CourseChoicesReviewComponent and its corresponding markup to also handle the error summary at the top of the page.

Reply:
Oh it goes into the error summary too, disregard then.